resolver_wrapper.go

```diff
@@ -79,6 +80,18 @@ func (ccr *ccResolverWrapper) start() error {
 			Authority:            ccr.cc.authority,
 			MetricsRecorder:      ccr.cc.metricsRecorderList,
 		}
+
+		// https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md
+		// defines CONNECTING as follows:
+		// - The channel is trying to establish a connection and is waiting to
+		//   make progress on one of the steps involved in name resolution, TCP
+		//   connection establishment or TLS handshake. This may be used as the
+		//   initial state for channels upon creation.
+		//
+		// We are starting the name resolver here as part of exiting IDLE, so
+		// transitioning to CONNECTING is the right thing to do.
```

Comment:
IMO comments should be short and to the point.

Short comments make the code take up less space, which makes it easier to read and understand.  Long comments make long functions extremely long and not fit on the page.

Honestly, I think a comment for this action isn't even necessary.  But if you think we need one, this could be:

```go
// Set state to CONNECTING before building the name resolver
// so the channel does not remain in IDLE.
```

Also, could this code (and the error handling below) go into `exitIdleMode` instead?  It might make more sense to be in the channel directly.

Reply:
Done.